extensions/cornerstone/src/initContextMenu.ts

```diff
@@ -0,0 +1,132 @@
+import { eventTarget, Types, EVENTS } from '@cornerstonejs/core';
+import { Enums } from '@cornerstonejs/tools';
+import { setEnabledElement } from './state';
+
+const cs3DToolsEvents = Enums.Events;
+
+const showContextMenuDefault: Types.CommandCustomization = {
+  commands: [
+    {
+      commandName: 'showViewerContextMenu',
+      commandOptions: {
+        menuName: 'cornerstoneContextMenu',
+      },
+      context: 'CORNERSTONE',
+    },
+  ],
+};
+
+function initContextMenu({
+  cornerstoneViewportService,
+  customizationService,
+  commandsManager,
+}): void {
+  /**
+   * Finds tool nearby event position triggered.
+   *
+   * @param {Object} commandsManager mannager of commands
+   * @param {Object} event that has being triggered
+   * @returns cs toolData or undefined if not found.
+   */
+  const findNearbyToolData = evt => {
+    if (!evt?.detail) {
+      return;
+    }
+    const { element, currentPoints } = evt.detail;
+    return commandsManager.runCommand(
+      'getNearbyToolData',
+      {
+        element,
+        canvasCoordinates: currentPoints?.canvas,
+      },
+      'CORNERSTONE'
+    );
+  };
+
+  const getShowContextMenu = (): Types.UICommand =>
+    customizationService.getModeCustomization(
+      'showContextMenu',
+      showContextMenuDefault
+    );
+
+  const showContextMenu = (contextMenuProps): void => {
+    customizationService.recordInteraction(
+      getShowContextMenu(),
+      contextMenuProps
+    );
+  };
+
+  const onRightClick = event => {
+    const nearbyToolData = findNearbyToolData(event);
+    showContextMenu({
+      event,
+      nearbyToolData,
+    });
```

Comment:
I honesetly think having this three functions refactored has added to complexity of reading it. It all can be under onRightClick

Reply:
It is more complex, but it allows defining and showing other types of context menus, and also allows other types of actions to be applied based on the incoming event.   For example, suppose I wanted right click to show the context menu, but control right click to bring up the labels dialog.  Binding that to a function like this allows defining a new command that can distinguish those two operations and run the right sub-commands to apply that.